tests/phpunit/test-class-wc-stripe-database-cache.php

```diff
@@ -0,0 +1,171 @@
+
+		$this->assertEquals( $data, $result );
+	}
+
+	/**
+	 * Test cache expiration.
+	 */
+	public function test_cache_expiration() {
+		$key  = 'expiring_key';
+		$data = 'expiring_data';
+
+		// Set cache with 1 hour TTL.
+		WC_Stripe_Database_Cache::set( $key, $data, HOUR_IN_SECONDS );
+
+		// Should be available immediately.
+		$this->assertEquals( $data, WC_Stripe_Database_Cache::get( $key ) );
+
+		// Update the in-memory-cache to simulate expiration.
+		$reflection = new ReflectionClass( 'WC_Stripe_Database_Cache' );
+		$property = $reflection->getProperty( 'in_memory_cache' );
+		$property->setAccessible( true );
+		$in_memory_cache = $property->getValue();
+		$in_memory_cache[ $key ]['updated'] -= HOUR_IN_SECONDS + 1; // Set update time to 1h 1s ago.
+		$property->setValue( null, $in_memory_cache );
+
+		// Should be expired.
+		$this->assertNull( WC_Stripe_Database_Cache::get( $key ) );
+
+		// Remove the in-memory-cache objects.
+		$property->setValue( null, [] );
+
+		// Update the database option to simulate expiration.
+		$cache_contents = get_option( $key );
+		$cache_contents['updated'] -= HOUR_IN_SECONDS + 1; // Set update time to 1h 1s ago.
+		update_option( $key, $cache_contents );
+
+		// Should be expired.
+		$this->assertNull( WC_Stripe_Database_Cache::get( $key ) );
+	}
+
+	/**
+	 * Test deleting from cache.
+	 */
+	public function test_delete_cache() {
+		$key  = 'delete_key';
+		$data = 'delete_data';
+
+		// Set and verify data is in cache.
+		WC_Stripe_Database_Cache::set( $key, $data );
+		$this->assertEquals( $data, WC_Stripe_Database_Cache::get( $key ) );
+
+		// Delete and verify data is removed.
+		WC_Stripe_Database_Cache::delete( $key );
+		$this->assertNull( WC_Stripe_Database_Cache::get( $key ) );
+	}
+
+	/**
+	 * Test in-memory cache.
+	 */
+	public function test_in_memory_cache() {
+		$key  = 'memory_key';
+		$data = 'memory_data';
+
+		// Set data.
+		WC_Stripe_Database_Cache::set( $key, $data );
+
+		// Get data once.
+		$result1 = WC_Stripe_Database_Cache::get( $key );
+
+		// Modify the option directly to check the second read in the same process uses the in-memory cache.
+		update_option( $key, null );
+
+		// Get data twice - second call should use in-memory cache.
+		$result2 = WC_Stripe_Database_Cache::get( $key );
+
+		$this->assertEquals( $data, $result1 );
+		$this->assertEquals( $data, $result2 );
+	}
+
+	/**
+	 * Test cache with different data types.
+	 */
+	public function test_cache_with_different_data_types() {
+		$test_cases = [
+			'string'  => 'test string',
+			'integer' => 123,
+			'array'   => [ 'key' => 'value' ],
+			'object'  => (object) [ 'property' => 'value' ],
+			'boolean' => true,
+			'null'    => null,
+		];
+
+		foreach ( $test_cases as $type => $data ) {
+			$key    = "test_{$type}";
+			WC_Stripe_Database_Cache::set( $key, $data );
+			$result = WC_Stripe_Database_Cache::get( $key );
+			$this->assertEquals( $data, $result, "Failed to cache {$type} data type" );
+		}
+	}
```

Comment:
Nit: It feels like using a dataProvider could make it slightly easier to "see" what failed from the test descriptions when things go wrong. I am 1000% in favour of using test runners with different inputs, so the quibble here is mostly around later ease in seeing failures from logs.
```suggestion
	/**
	 * Data provider for {@see test_cache_with_data_type()}.
	 *
	 * @return array Array of test cases.
	 */
	public function provide_data_type_test_cases() {
		return [
			'string'  => [ 'string', 'test string' ],
			'integer' => [ 'integer', 123 ],
			'array'   => [ 'array', [ 'key' => 'value' ] ],
			'object'  => [ 'object', (object) [ 'property' => 'value' ] ],
			'boolean' => [ 'boolean', true ],
			'null'    => [ 'null', null ],
		];
	}

	/**
	 * Test cache with different data types.
	 *
	 * @dataProvider provide_data_type_test_cases
	 */
	public function test_cache_with_different_data_types( $key_suffix, $data ) {
		$key    = "test_{$key_suffix}";
		WC_Stripe_Database_Cache::set( $key, $data );
		$result = WC_Stripe_Database_Cache::get( $key );
		$this->assertEquals( $data, $result, "Failed to cache and fetch data type" );
	}
```

Reply:
As a separate thought, it may be worth combining these data type tests with `test_cache_expiration()` and `test_in_memory_cache()`. I think it would be good to make sure that all layers are working correctly across all types.